src/app/page/userProfile/profile.component.ts

```diff
@@ -0,0 +1,104 @@
+import {Component, OnInit} from '@angular/core';
+import {Router} from '@angular/router';
+import {ProfileService} from './profile.service';
+import {UserDetailsDto} from '../../dto/UserDetailsDto';
+import {Person} from '../../dto/Person';
+import {AbstractControl, FormBuilder, FormGroup, Validators} from '@angular/forms';
+import {PasswordDTO} from '../../dto/PasswordDTO';
+import {SERVER_ADDRESS} from '../../services/orlp.settings';
+
+function passwordMatcher(c: AbstractControl) {
+  const passwordControl = c.get('password');
+  const confirmPassword = c.get('confirmPassword');
+  if (passwordControl.value === confirmPassword.value) {
+    return null;
+  }
+  return {'match': true};
+}
+
+@Component({
+  providers: [ProfileService],
+  templateUrl: ('./profile.component.html'),
+  styleUrls: ['profile.component.css']
+})
+export class ProfileComponent implements OnInit {
+
+  public userProfile: UserDetailsDto;
+  public firstName: string;
+  public lastName: string;
+  public person: Person = new Person();
+  public currentPassword: string;
+  public newPassword: string;
+  public showForm: boolean;
+  public showModal: boolean;
+  public chosenImage: boolean = false;
+  public authenticationType: string;
+  public imageUrl: string = SERVER_ADDRESS + 'api/private/user/profile/image';
```

Comment:
I believe, as you alredy know the right way to work with links, you could make as link in UserDetailsDto.